modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/PodcastGrouping.kt

```diff
@@ -81,7 +83,9 @@ sealed class PodcastGrouping(@StringRes val groupName: Int, val sortFunction: ((
      * @param episodes A sorted list of episodes
      * @return A pair of episodes and their group indexes
      */
-    open fun formGroups(episodes: List<PodcastEpisode>, reversedSort: Boolean = false, resources: Resources): List<List<PodcastEpisode>> {
+    open fun formGroups(episodes: List<PodcastEpisode>, podcast: Podcast, resources: Resources): List<List<PodcastEpisode>> {
+        val reversedSort = podcast.podcastGrouping is Season &&
+            podcast.episodesSortType == EpisodesSortType.EPISODES_SORT_BY_DATE_DESC
```

Comment:
I just moved this reverseSort logic into this function because (1) we were using the same logic everywhere we called this function and (2) it seemed like not super-intuitive logic so at least now we only have this not-intuitive-logic in a single place instead of multiple places.